crypto/evp/signature.c

```diff
@@ -387,16 +502,20 @@ const OSSL_PARAM *EVP_SIGNATURE_settable_ctx_params(const EVP_SIGNATURE *sig)
     return sig->settable_ctx_params(NULL, provctx);
 }
 
-static int evp_pkey_signature_init(EVP_PKEY_CTX *ctx, int operation,
-                                   const OSSL_PARAM params[])
+static int evp_pkey_signature_init(EVP_PKEY_CTX *ctx, EVP_SIGNATURE *signature,
```

Comment:
This function was already fairly hard to digest before the changes. if there is a way of breaking it into a few functions perhaps that would be a little better.

Reply:
A refactoring could be done, but I'm not sure it should be part of this PR...  that's not going to make things easier to digest in terms of a diff...